### PR TITLE
Test: Added Integration tests for `HExit.Success`

### LINKED ISSUE #829: Add Integration tests when Http executes to `HExit.success`
**What is not easy to do right now?**
Current integration tests, not cover the scenarios when Http evaluates to `HExit.success`

**Describe the solution you'd like**
Add a static app with `Http.collect`
test for Get and Post requests for success, failure and Not found.

**Why is it important?**
It will help us catch regression issues

### PR DESCRIPTION
**Pull request by zsfVishnu-d11**
fixes #829

### CODE REVIEW

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala [hunk 1]
```diff
@@ -262,4 +269,32 @@ object ServerSpec extends HttpRunnableSpec {
...
+
+  def validationAppSpec = suite("ValidationAppSpec") {
+    testM("200 response GET") {
+      val actual = status(!! / "HExitSuccess")
+      assertM(actual)(equalTo(Status.OK))
+    } +
+      testM("200 response POST") {
+        val actual = status(!! / "HExitSuccess", Method.POST)
+        assertM(actual)(equalTo(Status.OK))
+      } +
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/service/ServerSpec.scala]: combine these two test using `HttpGen.method` and `checkAllM`

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala [hunk 2]
```diff
@@ -262,4 +269,32 @@ object ServerSpec extends HttpRunnableSpec {
...
+      testM("400 response GET") {
+        val actual = status(!! / "HExitFailure")
+        assertM(actual)(equalTo(Status.BAD_REQUEST))
+      } +
+      testM("400 response POST") {
+        val actual = status(!! / "HExitFailure", Method.POST)
+        assertM(actual)(equalTo(Status.BAD_REQUEST))
+      } +
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/service/ServerSpec.scala]: These can be combined similarly

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala [hunk 3]
```diff
@@ -262,4 +269,32 @@ object ServerSpec extends HttpRunnableSpec {
...
+      testM("404 response GET") {
+        val actual = status(!! / "A")
+        assertM(actual)(equalTo(Status.NOT_FOUND))
+      } +
+      testM("404 response POST") {
+        val actual = status(!! / "A", Method.POST)
+        assertM(actual)(equalTo(Status.NOT_FOUND))
+      }
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/service/ServerSpec.scala]: we can keep only one of these

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala (hunk 1 above)
Comment (zsfVishnu-d11) [reply zio-http/src/test/scala/zhttp/service/ServerSpec.scala]: done

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala (hunk 2 above)
Comment (zsfVishnu-d11) [reply zio-http/src/test/scala/zhttp/service/ServerSpec.scala]: done

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala (hunk 3 above)
Comment (zsfVishnu-d11) [reply zio-http/src/test/scala/zhttp/service/ServerSpec.scala]: done

File: zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala [hunk 4]
```diff
@@ -43,12 +43,12 @@ abstract class HttpRunnableSpec extends DefaultRunnableSpec { self =>
     } yield response
   }
 
-  def status(path: Path): HttpIO[Any, Status] = {
+  def status(path: Path, method: Method = Method.GET): HttpIO[Any, Status] = {
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala]:
```suggestion
  def status(method: Method = Method.GET, path: Path): HttpIO[Any, Status] = {
```

Comment (zsfVishnu-d11) [reply zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala]: done

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala [hunk 5]
```diff
@@ -31,7 +31,12 @@ object ServerSpec extends HttpRunnableSpec {
     case Method.GET -> !! / "get%2Fsuccess" => ZIO.succeed(Response.ok)
   }
 
-  private val app = serve { staticApp ++ DynamicServer.app }
+  private val validationApp: Http[Any, Nothing, Request, Response] = Http.collect[Request] {
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/service/ServerSpec.scala]:
```suggestion

  // Use this route to test anything that doesn't require ZIO related computations.
  private val nonZIO: Http[Any, Nothing, Request, Response] = Http.collect[Request] {
```

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala [hunk 6]
```diff
@@ -241,25 +246,47 @@ object ServerSpec extends HttpRunnableSpec {
...
       } +
       testM("Multiple 200 response") {
         for {
-          data <- status(!! / "success").repeatN(1024)
+          data <- status(path = !! / "success").repeatN(1024)
         } yield assertTrue(data == Status.OK)
       }
   }
+
+  def validationAppSpec = suite("ValidationAppSpec") {
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/service/ServerSpec.scala]:
```suggestion
  def nonZIOSpec = suite("NonZIOSpec") {
```

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala (hunk 5 above)
Comment (zsfVishnu-d11) [reply zio-http/src/test/scala/zhttp/service/ServerSpec.scala]: done

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala (hunk 6 above)
Comment (zsfVishnu-d11) [reply zio-http/src/test/scala/zhttp/service/ServerSpec.scala]: done

**Review by tusharmath — APPROVED** [review]

**Pull request merged**